src/Cli/dotnet/CommandLineParser.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.CommandLine;
+using System.CommandLine.Builder;
+using System.CommandLine.Help;
+using System.CommandLine.Invocation;
+using System.CommandLine.IO;
+using System.CommandLine.Parsing;
+using System.Reflection;
+using Microsoft.DotNet.Tools.Help;
+using Microsoft.DotNet.Tools.MSBuild;
+
+namespace Microsoft.DotNet.Cli
+{
+    internal static class CommandLineParser
+    {
+        private static readonly Parser Parser;
+        private static readonly RootCommand RootCommand = new RootCommand();
+
+        // Subcommands
+        private static readonly Command[] Subcommands = new Command[]
+        {
+            AddCommandParser.GetCommand(),
+            BuildCommandParser.GetCommand(),
+            BuildServerCommandParser.GetCommand(),
+            CleanCommandParser.GetCommand(),
+            CompleteCommandParser.GetCommand(),
+            FsiCommandParser.GetCommand(),
+            ListCommandParser.GetCommand(),
+            MSBuildCommandParser.GetCommand(),
+            NewCommandParser.GetCommand(),
+            NuGetCommandParser.GetCommand(),
+            PackCommandParser.GetCommand(),
+            ParseCommandParser.GetCommand(),
+            PublishCommandParser.GetCommand(),
+            RemoveCommandParser.GetCommand(),
+            RestoreCommandParser.GetCommand(),
+            RunCommandParser.GetCommand(),
+            SlnCommandParser.GetCommand(),
+            StoreCommandParser.GetCommand(),
+            TestCommandParser.GetCommand(),
+            ToolCommandParser.GetCommand(),
+            VSTestCommandParser.GetCommand(),
+            HelpCommandParser.GetCommand()
+        };
+
+        // Internal commands
+        public static readonly Command InstallSuccessCommand = InternalReportinstallsuccessCommandParser.GetCommand();
+
+        // Global options
+        public static readonly Option DiagOption = new Option(new[] { "-d", "--diagnostics" })
+        {
+            IsHidden = true
+        };
+
+        // Informational options
+        public static readonly Option VersionOption = new Option("--version");
```

Comment:
Though the output might be slightly different, this is supported directly by System.CommandLine.